modules/geo/src/main/java/org/opensearch/geo/algorithm/PolygonGenerator.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.algorithm;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.awt.geom.Point2D;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.IntStream;
+
+/**
+ * Helper class to generate a polygon. Keeping this in the src folder so that GeoSpatial plugin can take advantage of
+ * this helper to create the Polygons, rather than hardcoding the values.
+ */
+public class PolygonGenerator {
+
+    private static final Logger LOG = LogManager.getLogger(PolygonGenerator.class);
+
+    /**
+     * A helper function to create the Polygons for testing. The returned list of double array where first element
+     * contains all the X points and second contains all the Y points.
+     *
+     * @param xPool a {@link java.util.List} of {@link Double}
+     * @param yPool a {@link java.util.List} of {@link Double}
+     * @return a {@link List} of double array.
+     */
+    public static List<double[]> generatePolygon(final List<Double> xPool, final List<Double> yPool, final Random random) {
+        if (CollectionUtils.isEmpty(xPool) || CollectionUtils.isEmpty(yPool)) {
```

Comment:
```suggestion
        if (xPool.isEmpty() == true || yPool.isEmpty() == true) {
```

Reply:
The suggested change can lead to NPE if the xPool or yPool is null.

Reply:
Let's add the null check?

Reply:
The new change has already null checks in it.